stdlib/SuiteSparse/src/umfpack.jl

```diff
@@ -406,6 +406,30 @@ for itype in UmfpackIndexTypes
                         mx, mz, C_NULL, lu.numeric, umf_info)
             complex(mx[], mz[])
         end
+        function logabsdet(F::UmfpackLU{T, $itype}) where T # return log(abs(det)) and sign(det)
+            n = checksquare(F)
+            issuccess(F) || return log(zero(real(T))), log(one(T))
+            U = diag(F.U)
+            Rs = F.Rs
+            p = F.p
+            q = F.q
+            c = 0
+            P = one(T)
+            abs_det = zero(real(T))
+            @inbounds for i in 1:n
+                dg_ii = U[i] / Rs[i]
+                P *= sign(dg_ii)
+                if p[i] != i
+                    c += 1
+                end
+                if q[i] != i
+                    c += 1
+                end
```

Comment:
`p` and `q` play different roles than `F.ipiv` in the generic method, that's why the sign computation is wrong here. I couldn't figure out how to reconstruct `ipiv` from `p`, but the following is a way to compute the sign of the permutations:
```suggestion
                for j in i+1:n
                    if p[i] > p[j]
                        c += 1
                    end
                    if q[i] > q[j]
                        c += 1
                    end
                end
```
Hopefully, that would fix the test failures.

Reply:
Thanks, was just about to dig into this myself!